pkg/store/prometheus.go

```diff
@@ -496,9 +499,39 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		sers []map[string]string
+		err  error
+	)
+
+	vals := []string{}
+	version := p.promVersion()
+
+	if len(r.Matchers) == 0 || version > "2.24" {
+		vals, err = p.client.LabelValuesInGRPC(ctx, p.base, r.Label, r.Matchers, r.Start, r.End)
```

Comment:
Are we making gRPC call?

Reply:
YES

Reply:
The naming makes me confused, too. We are making HTTP calls but return GRPC error.